src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/refdatafixes/ReferenceDataFixesService.java

```diff
@@ -0,0 +1,123 @@
+        String requiredJudgeCode = refDataFixesData.getRequiredJudgeCode();
+        String caseTypeId = refDataFixesDetails.getCaseTypeId().contains("_") ?
+                Strings.split(refDataFixesDetails.getCaseTypeId(), "_")[0] : refDataFixesDetails.getCaseTypeId();
+        List<String> dates = getDateRangeForSearch(refDataFixesDetails);
+        String dateFrom = dates.get(0);
+        String dateTo = dates.get(1);
+        try {
+            List<SubmitEvent> submitEvents = dataSource.getData(caseTypeId, dateFrom, dateTo, ccdClient);
+            if (CollectionUtils.isNotEmpty(submitEvents)) {
+                log.info(CASES_SEARCHED + submitEvents.size());
+                for (SubmitEvent submitEvent : submitEvents) {
+                    CaseData caseData = submitEvent.getCaseData();
+                    setJudgeName(caseData, existingJudgeCode, requiredJudgeCode);
+
+                    CCDRequest returnedRequest = ccdClient.startEventForCase(authToken, caseTypeId,
+                            refDataFixesDetails.getJurisdiction(), String.valueOf(submitEvent.getCaseId()));
+                    ccdClient.submitEventForCase(authToken, caseData, caseTypeId,
+                            refDataFixesDetails.getJurisdiction(), returnedRequest, String.valueOf(submitEvent.getCaseId()));
+                }
+                log.info(String.format(
+                        "Existing Judge's code in all cases from %s to %s is " +
+                                "updated to required judge's code", dateFrom, dateTo));
+            }
+            return refDataFixesData;
+        } catch (Exception ex) {
+            log.error(MESSAGE + refDataFixesDetails.getCaseId(), ex);
+            throw new CaseRetrievalException(MESSAGE + refDataFixesDetails.getCaseId() + ex.getMessage());
+        }
+    }
+
+    private List<String> getDateRangeForSearch(RefDataFixesDetails refDataFixesDetails) {
+        var refDataFixesData = refDataFixesDetails.getCaseData();
+        boolean isRangeHearingDateType = refDataFixesData.getHearingDateType().equals(RANGE_HEARING_DATE_TYPE);
+        String dateFrom;
+        String dateTo;
+        if (!isRangeHearingDateType) {
+            dateFrom = LocalDate.parse(refDataFixesData.getDate(), OLD_DATE_TIME_PATTERN2)
+                    .atStartOfDay().format(OLD_DATE_TIME_PATTERN);
+            dateTo = LocalDate.parse(refDataFixesData.getDate(), OLD_DATE_TIME_PATTERN2)
+                    .atStartOfDay().plusDays(1).minusSeconds(1).format(OLD_DATE_TIME_PATTERN);
+        } else {
+            dateFrom = LocalDate.parse(refDataFixesData.getDateFrom(), OLD_DATE_TIME_PATTERN2)
+                    .atStartOfDay().format(OLD_DATE_TIME_PATTERN);
+            dateTo = LocalDate.parse(refDataFixesData.getDateTo(), OLD_DATE_TIME_PATTERN2)
+                    .atStartOfDay().plusDays(1).minusSeconds(1).format(OLD_DATE_TIME_PATTERN);
+        }
+        return new ArrayList<>(List.of(dateFrom, dateTo));
```

Comment:
You don't need to create an ArrayList. What is wrong with returning the value from List.of ?

Reply:
returning list.of only now